openvm/src/powdr_extension/plonk/chip.rs

```diff
@@ -104,7 +93,9 @@ impl<PB: ProverBackend<Matrix = DenseMatrix<BabyBear>>> Chip<BabyBearSC, PB> for
             .enumerate()
             .map(|(index, c)| (c.id, index))
             .collect();
-        let witness = self.executor.generate_witness();
+        let witness = self
```

Comment:
shared trace generation for single row and plonk. plonk takes this trace and builds the plonk trace from it.